Dockerfile

```diff
@@ -1,29 +1,35 @@
-FROM python:3.6-slim
-
-SHELL ["/bin/bash", "-c"]
-
-RUN apt-get update -qq && \
-  apt-get install -y --no-install-recommends \
-  build-essential && \
-  apt-get clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
-  mkdir /app
-
-WORKDIR /app
-
-# Copy as early as possible so we can cache ...
-COPY requirements.txt .
-
-RUN pip install -r requirements.txt --no-cache-dir
-
-COPY . .
-
-RUN pip install -e . --no-cache-dir
+FROM python:3.6-alpine as python_builder
+RUN apk update && \
+  apk add \
+  build-base \
+  curl \
+  git
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH "/root/.poetry/bin:/opt/venv/bin:${PATH}"
+
+COPY poetry.lock pyproject.toml /opt/project/
+RUN python -m venv /opt/venv && \
+  source /opt/venv/bin/activate && \
+  pip install -U pip && \
+  cd /opt/project && \
+  poetry install --no-dev --no-interaction
+
+COPY . /opt/project
+RUN source /opt/venv/bin/activate && \
+  cd /opt/project && \
+  poetry install --no-dev --no-interaction
+
+FROM python:3.6-alpine
```

Comment:
Alpine is a different linux distribution, so it's different when it comes to package management. However, it's more minimal than `slim` (it's 135Mb vs 200Mb with the current config).
I'll change it back to `slim` in the next commit though — the difference is probably not that big to change it.

Reply:
alright, I think that makes sense - even though I agree it would be great to shave off that additional bit of image size I'd could create some problems :+1: